src/js/game/LevelMVC/LevelModel.js

```diff
@@ -68,19 +68,44 @@ module.exports = class LevelModel {
       this.controller.player = this.player;
 
       if (levelData.useAgent) {
-        this.usingAgent = levelData.useAgent;
-        [x, y] = levelData.agentStartPosition;
-        const startingBlock = this.actionPlane.getBlockAt(levelData.agentStartPosition);
-        this.agent = new Agent(this.controller, "PlayerAgent", x, y, "Agent", !startingBlock.getIsEmptyOrEntity(), levelData.agentStartDirection);
-        this.controller.levelEntity.pushEntity(this.agent);
-        this.controller.agent = this.agent;
+        this.doAgentSpawn(levelData);
       }
     }
 
     this.computeShadingPlane();
     this.computeFowPlane();
   }
 
+  doAgentSpawn(levelData = null, aX = -1, aY = -1) {
```

Comment:
something like `agentSpawn(levelData, positionOverride, directionOverride)` would be a lot cleaner. It combines the X and Y into one position argument, which is semantically more accurate, adds the direction so we don't have to hardcode it, and also means that all arguments will default to `undefined` if not specified, which is again semantically more accurate than `null` or `-1`

Reply:
Oh right, because you don't need to give explicitly defined default arguments in JS. 😰 

I can change that for sure. I'm just in general not in love with having to add a case to the function that is -only- used for one special unique level but... I can make this cleaner.